crates/contract/src/impersonate.rs

```diff
@@ -0,0 +1,146 @@
+use alloy_network::Network;
+use alloy_primitives::U256;
+use alloy_provider::{ext::AnvilApi, Provider};
+use alloy_rpc_types_eth::TransactionRequest;
+use alloy_transport::TransportError;
+use futures::try_join;
+
+/// A utility for impersonating an Ethereum account to send transactions using Anvil.
+///
+/// This helper simplifies the process of:
+/// 1. Impersonating an account
+/// 2. Optionally funding it with ETH
+/// 3. Sending a transaction from it without a signer
+/// 4. Stopping the impersonation
+///
+/// # Example
+///
+/// ```no_run
+/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+///
+/// use alloy_network::TransactionBuilder;
+/// use alloy_primitives::address;
+/// use alloy_provider::{Provider, ProviderBuilder};
+/// use alloy_rpc_types_eth::TransactionRequest;
+///
+/// let provider = ProviderBuilder::new().connect_anvil();
+///
+/// let from = address!("0x000000000000000000000000000000000000dEaD");
+/// let to = address!("0x000000000000000000000000000000000000beef");
+/// let tx = TransactionRequest::default().with_from(impersonate).with_to(to).with_value(val);
+///
+/// let call = ImpersonatedCall::new(provider, tx, Some(U256::from(1e18 as u64)));
+/// let receipt = call.send_impersonated_tx().await?;
+/// println!("Impersonated tx succeeded in block: {:?}", receipt.block_number);
+/// # Ok(())
+/// # }
+/// ```
+
+#[derive(Debug, Clone)]
+pub struct ImpersonatedCall<P, N>
```

Comment:
actually, maybe we dont need a dedicated call for this and instead can add this as a utility trait function to AnvilAp, that takes the request and a ```struct ImpersonateConfig { fund, stop_impersonate}```